frontend/src/pages/Overview/OverviewPage.tsx

```diff
@@ -848,18 +866,13 @@ export class OverviewPageComponent extends React.Component<OverviewProps, State>
 
       if (
         serverConfig.kialiFeatureFlags.istioUpgradeAction &&
-        serverConfig.istioCanaryRevision.upgrade &&
-        serverConfig.istioCanaryRevision.current
+        this.state.canaryUpgradeStatus?.upgradeVersion &&
+        this.state.canaryUpgradeStatus.currentVersion
```

Comment:
Instead of picking between two options "upgrade" / "downgrade", IMO this should change to pick between any one of the revisions that are managing this cluster. A picker more like:
```
1-21-0
1-22-0
experimental
```

Reply:
You can find the clusters that a controlplane manages by looking at `ControlPlane.ManagedClusters` https://github.com/kiali/kiali/blob/2a0f9deda6b93369363e04ea1c381113c8b9de18/models/mesh.go#L42-L44